modo/file_utils.py

```diff
@@ -0,0 +1,28 @@
+"""Utilities to manage data files associated with model instances"""
+
+from pysam import AlignmentFile
+from smoc_schema.datamodel import DataEntity
+from typing import Mapping, List
+
+from .cram import extract_cram_metadata
+
+
+def extract_metadata(instance) -> List:
+    """Extract metadata from files associated to a model instance"""
+    if (
+        isinstance(instance, DataEntity)
```

Comment:
Thanks a lot - a very good suggestion. Just in case and to explain:
This means that `modo.enrich_metadata()` needs to filter for instances that fulfill this condition before extracting metadata (before I just looped through all instances and returned an empty list, if there was no cram file associated).
I added a one-liner to do so in `modo.enrich_metadata()`. This should also be more efficient and keeps a more coherent logic in `extract_metadata()`